### PR TITLE
Fix README spelling and formatting for Synapse

### PR DESCRIPTION
**Pull request by annelo-msft**
Addresses https://github.com/azure/azure-sdk-for-net/issues/24069 for Synapse.

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -234,6 +234,13 @@
         "Undelete"
       ]
     },
+    {
+      "filename": "**/sdk/synapse/**/*.md",
+      "words": [
+        "dotnetcli",
```
Comment (heaths) [inline .vscode/cspell.json:240]: Curious how this is used. Links (even markdown links) shouldn't matter for spelling, and if used in text, this doesn't seem an appropriate marketing term. If it is used in a general, conforming manner, maybe it should be global to the repo?

File: sdk/synapse/Azure.Analytics.Synapse.AccessControl/README.md [hunk 2]
```diff
@@ -98,8 +108,8 @@ RoleDefinitionsClient definitionsClient = new RoleDefinitionsClient(new Uri(endp
 First, you need to the determine the ID of the role you wish to assign, along with the ID of the principal you wish to assign that role.
 
 ```C# Snippet:PrepCreateRoleAssignment
-Response roleDefinitionsReponse = definitionsClient.GetRoleDefinitions(new());
-BinaryData roleDefinitionsContent = roleDefinitionsReponse.Content;
+Response roleDefinitionsResponse = definitionsClient.GetRoleDefinitions(new());
```
Comment (heaths) [inline sdk/synapse/Azure.Analytics.Synapse.AccessControl/README.md:111]: Given this is extracted, the source would have to be updated as well (or was it wrong and just never flagged)?

**Review by heaths — APPROVED** [review]
Couple of questions/concerns, but nothing blocking.

Also, was it part of the issue you're addressing already to add in - when done across services - the Microsoft Docs Pack as a recommended extension? As noted in Teams today, we may want to hold off on that until the bug is fixed: https://github.com/DavidAnson/vscode-markdownlint/issues/204#issuecomment-963642766

File: .vscode/cspell.json (hunk 1 above)
Comment (annelo-msft) [reply .vscode/cspell.json]: You're right, it's repo-wide.  I'll make it global - thanks!

(For e.g. here: https://raw.githubusercontent.com/Azure/azure-sdk-for-net/main/sdk/storage/Azure.Storage.Common/README.md

image)

**Comment by annelo-msft** [comment]
> Also, was it part of the issue you're addressing already to add in - when done across services - the Microsoft Docs Pack as a recommended extension? 

I'll just address Synapse for now.

File: .vscode/cspell.json (hunk 1 above)
Comment (heaths) [reply .vscode/cspell.json]: Ah. I completely forgot about that. Someone did it a few months ago because it renders better on docs.microsoft.com.

**Pull request merged**